lib/logger.js

```diff
@@ -174,22 +175,19 @@ Logger.prototype._levelToMatcher = function _levelToMatcher(level) {
  * If the message should be logged, returns an object with bunyan log level
  * and a specialized logger for the given component, otherwise returns undefined
  *
- * @param {String} level a logging level
+ * @param {String} levelPath a logging level + component ( e.g. 'warn/component_name' )
  * @returns {Object|undefined} corresponding bunyan log level and a specialized logger
  * @private
  */
-Logger.prototype._getComponentLogConfig = function(level) {
-    if (level && Object.keys(this._log_components).length) {
-        var slashIndex = level.indexOf('/');
-        if (slashIndex > 0) {
-            var component = level.substr(slashIndex + 1);
-            var logProbability = this._log_components[component];
-            if (logProbability && Math.ceil(Math.random() * 1024) < logProbability) {
-                return {
-                    level: level.substring(0, slashIndex),
-                    logger: this._componentLoggers[component]
-                };
-            }
+Logger.prototype._getComponentLogConfig = function(levelPath) {
+    var logProbability = this._log_components[levelPath];
+    if (logProbability && Math.random() < logProbability) {
+        var levelMatch = this._allLevelsMatcher.exec(levelPath);
```

Comment:
Is this check really needed? `_allLevelsMatcher` is set to `trace`, so this will always be true.